boards/common/stm32/include/cfg_timer_tim2_and_tim5.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_common_stm32
+ * @{
+ *
+ * @file
+ * @brief       Common configuration for STM32 Timer peripheral based on TIM2
+ *              and TIM5
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef CFG_TIMER_TIM2_AND_TIM5_H
+#define CFG_TIMER_TIM2_AND_TIM5_H
+
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name   Timer configuration
+ * @{
+ */
+static const timer_conf_t timer_config[] = {
+    {
+        .dev      = TIM2,
+#if defined(CPU_FAM_STM32L0) || defined(CPU_FAM_STM32L1)
+        .max      = 0x0000ffff,
+#else
+        .max      = 0xffffffff,
+#endif
+#if defined(CPU_FAM_STM32L4) || defined(CPU_FAM_STM32WB) || \
+    defined(CPU_FAM_STM32WL) || defined(CPU_FAM_STM32G4)
+        .rcc_mask = RCC_APB1ENR1_TIM2EN,
+#elif CPU_FAM_STM32MP1
+        .rcc_mask = RCC_MC_APB1ENSETR_TIM2EN,
+#else
+        .rcc_mask = RCC_APB1ENR_TIM2EN,
+#endif
+        .bus      = APB1,
+        .irqn     = TIM2_IRQn
+    },
+    {
+        .dev      = TIM5,
+        .max      = 0xffffffff,
```

Comment:
Is there any reason for the sorting?
If TIM5 is always available and always 32bit, it should be the first in the array, to trickle down as default for ztimer_used.

Reply:
At least on the L4 it indeed seems to be available and 32 bit. I will swap the two.

Reply:
https://github.com/RIOT-OS/RIOT/blob/1961274705d75aee1c7482c19493b053740de150/boards/nucleo-l152re/doc.txt#L29

Yep, at least on STM32L1 the sorting helps ztimer